crates/nu-cmd-lang/src/core_commands/describe.rs

```diff
@@ -245,39 +252,42 @@ fn run(
             if !options.detailed {
                 Value::string(value.get_type().to_string(), head)
             } else {
-                describe_value(value, head, engine_state)
+                describe_value(&value, head, engine_state)
             }
         }
     };
 
     Ok(description.into_pipeline_data())
 }
 
-fn compact_primitive_description(mut value: Value) -> Value {
-    if let Value::Record { ref mut val, .. } = value {
-        if val.len() != 1 {
-            return value;
-        }
-        if let Some(type_name) = val.to_mut().get_mut("type") {
-            return std::mem::take(type_name);
+enum Description {
+    String(String),
+    Record(Record),
+}
+
+impl Description {
+    fn into_value(self, span: Span) -> Value {
+        match self {
+            Description::String(ty) => Value::string(ty, span),
```

Comment:
Is this doing the same thing, of not using a single entry `{type: inner}` record if used in the context of enumerating primitive inner types where we currently just return the type string?

Reply:
If I understand you correctly, yes. Instead of returning `{type: "type"}` for primitives and then re-extracting out the type string, we now return `Description::String`. If we encounter `Description::String` right before the final top level return (line 279), only then do we convert it to a record value with the form `{type: "type"}`.

Reply:
Ah I got confused by the operations in the root `describe_value`. Thanks for clarifying and simplifying